src/hpack.erl

```diff
@@ -113,7 +114,8 @@ decode(Bin, Context) ->
 -spec decode(binary(), headers(), context())
             ->
                     {ok, {headers(), context()}}
-                  | {error, compression_error}.
+                  | {error, compression_error}
+                  | {error, {compression_error, {bad_header_packet, binary()}}}.
```

Comment:
I'm not sure that `compression_error` as a reason for an error is very helpful, unless we are always going to handle `compression_error` in the same way. Perhaps it could be `{error, {bad_header_packet, binary()}}` if we want to differentiate between different `compression_error` reasons.  If we want to be able to only match on `compression_error` but format different compression errors then I guess we do need `{error, {compression_error, _}}`. Either way I think we should bring the other compression errors into the same structure.